Makefile.dep

```diff
@@ -691,9 +699,24 @@ USEPKG += nanocoap
 USEMODULE += gnrc_sock_udp
 endif
 
+# all boards use gpio for leds
+FEATURES_OPTIONAL += periph_gpio
+
+# always select power management if available
+FEATURES_OPTIONAL += periph_pm
+
 # include package dependencies
 -include $(USEPKG:%=$(RIOTPKG)/%/Makefile.dep)
 
+# remove required features from optional features
+FEATURES_OPTIONAL := $(filter-out $(FEATURES_REQUIRED),$(FEATURES_OPTIONAL))
+
+# add all optional but provided features to the list of used features
+FEATURES_USED := $(sort $(FEATURES_REQUIRED) $(filter $(FEATURES_OPTIONAL),$(FEATURES_PROVIDED)))
+
+# all periph features correspond to a periph submodule
+USEMODULE += $(filter periph_%,$(FEATURES_USED))
```

Comment:
this looks off -> doesn't this possibly add something like `periph_cpp`? So shouldn't there be here (or above) some filter to restrict this to `periph_x` stuff only (as in `$(filter periph_%, $FEATURExxx)`)?

Reply:
(addressed offline)